src/Mayaqua/Network.c

```diff
@@ -11785,7 +11789,15 @@ bool AddChainSslCert(struct ssl_ctx_st *ctx, X *x)
 
 	if (x_copy != NULL)
 	{
-		SSL_CTX_add_extra_chain_cert(ctx, x_copy->x509);
+		if (x_copy->root_cert)
+		{
+			X509_STORE* store = SSL_CTX_get_cert_store(ctx);
+			X509_STORE_add_cert(store, x->x509);
+		}
+		else
+		{
+			SSL_CTX_add_extra_chain_cert(ctx, x_copy->x509);

```

Comment:
is it destroyed/freed somewhere ?

Reply:
I've added an X509_Free here, but I'm not sure there was actually a memory leak. I ran with valgrind after that though, no memory leaks detected.

Reply:
yes, valgrind is good. what happens if we wish to add another root chain instead of loaded one ? it's not possible, we can only add chains ?


as for memleak, I'd suggest to try clang asan as well. it might be enforced by setting the following variables before running ./configure

`export CC=clang CFLAGS="-g -fsanitize=address" LDLAGS="-g -fsanitize=address"`

Reply:
Sorry I didn't understand the question. I guess SSL_CTX_add_extra_chain_cert can work with any number of root certs currently loaded?

Reply:
yeah, but we can only add chains ? or we can also remove chains ?

Reply:
It reloads each new request. Yeah, we can't change it mid-request, but at each new SSL context creation they are reread from the file system.

Reply:
if we reload new certificate chain, it won't be leak ?

Reply:
It shouldn't, I see no reason why it should. Theese aren't cached anywhere and they get freed with the SSL context itself...